File: issta2018artifacts/minimized-bugs/buggy-clousot/PlumeLib/FnInverse.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.Contracts;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

class Arrays
{
    public static void Fill<T>(T[] a, T v)
    {
        for (int i = 0; i < a.Length; ++i)
            a[i] = v;
    }
}

public class FnInverse
{

    public static int[] fn_inverse(int[] a, int arange)
    {
        Contract.Requires(arange >= 0);
        int[] result = new int[arange];
        Arrays.Fill(result, -1);
        for (int i = 0; i < a.Length; i++)
        {
            int ai = a[i];
            if (ai != -1)
            {
                //:: error: array.access.unsafe.high error: array.access.unsafe.low
                if (result[ai] != -1)
                {
                    throw new InvalidOperationException("Not invertible");
                }
                //:: error: array.access.unsafe.high error: array.access.unsafe.low
                result[ai] = i;
            }
        }
        return result;
    }
}

File: issta2018artifacts/minimized-bugs/buggy-key/FnInverse.java
// simplified version of the plume-lib bug fixed in 0abfea178a813c3895fecebcbb8bcbed1cee1b42

public class FnInverse {

	/*@ public normal_behavior
@ requires arange >= 0;
@ ensures true;
@*/
    public static int[] fn_inverse(int[] a, int arange) {
	int[] result = new int[arange];
	for (int i = 0; i < a.length; i++) {
	    int ai = a[i];
	    if (ai != -1) {
		//:: error: array.access.unsafe.high error: array.access.unsafe.low
		if (result[ai] != -1) {
			return null;
		}
		//:: error: array.access.unsafe.high error: array.access.unsafe.low
		result[ai] = i;
	    }
	}
	return result;
    }
}
